lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -431,10 +569,69 @@ sub add_batch_job {
     return $results;
 }
 
-
+$json_schemas{get_batch_job_result} = joi->object->strict->props(
+    batch_id => $zm_validator->batch_id->required
+);
 sub get_batch_job_result {
-    my ( $self, $batch_id ) = @_;
-
+    my ( $self, $params ) = @_;
+    my $batch_id = "";
+    if (ref \$params eq "SCALAR") {
+        $batch_id = $params;
+    } else {
+        $batch_id = $params->{"batch_id"};
+    }
     return $self->{db}->get_batch_job_result($batch_id);
 }
+
+my $rpc_request = joi->object->props(
+    jsonrpc => joi->string->required,
+    id => joi->integer->min(0)->required,
+    method => joi->string->regex("[a-zA-Z0-9_-]*")->required
+);
+sub jsonrpc_validate {
```

Comment:
Maybe this should be mored to the Validator module? What do you think?

Reply:
Done, good idea.